linux-setup.sh

```diff
@@ -0,0 +1,253 @@
+#!/bin/bash -e
+
+# This is a script to perform build and install processes for the `haiku-format` tool on a
+# Linux platform. Run the script with no arguments to get usage. See the accompanying
+# `linux.md` file for instructions on general use.
+
+set -e
+set -o pipefail
+
+script_bin=$0
+build_dir="llvm-project/build"
+install_dir="/opt/haiku-format"
+llvm_version=18.1.8
+llvm_project=llvm-project
+llvm_base_url="https://github.com/llvm/${llvm_project}/releases/download/llvmorg-${llvm_version}"
+
+function hf_usage()
+{
+  cat <<EOF
+build.sh
+
+This script is designed to build the `haiku-format` tool and install it on an
+`apt`-based Linux system.
```

Comment:
```suggestion
This script is designed to build the 'haiku-format' tool and install it on an
'apt'-based Linux system.
```
The backquotes failed on macOS for me. Also, single quotes are consistent with the message in `hf_main()`.